src/sinks/gcp/stackdriver_logging.rs

```diff
@@ -268,31 +223,16 @@ mod tests {
         "#,
         )
         .unwrap();
-        let request = config.write_request();
-        request
-            .find("\"entries\":[]")
-            .expect("Could not find required entries list");
-    }
 
-    #[test]
-    fn encode_valid1() {
-        let log = LogEvent::from_iter([("message", "hello world")].iter().map(|&s| s));
-        let body = encode_event(log.into(), &Encoding::Default.into());
-        let body = String::from_utf8_lossy(&body);
-        assert_eq!(body, "{\"jsonPayload\":{\"message\":\"hello world\"}},");
-    }
+        let sink = StackdriverSink {
+            config,
+            creds: None,
+        };
 
-    #[test]
-    fn encode_valid2() {
-        let log1 = LogEvent::from_iter([("message", "hello world")].iter().map(|&s| s));
-        let log2 = LogEvent::from_iter([("message", "killroy was here")].iter().map(|&s| s));
-        let mut event = encode_event(log1.into(), &Encoding::Default.into());
-        event.extend(encode_event(log2.into(), &Encoding::Default.into()));
-        let body = String::from_utf8_lossy(&event);
-        assert_eq!(
-            body,
-            "{\"jsonPayload\":{\"message\":\"hello world\"}},{\"jsonPayload\":{\"message\":\"killroy was here\"}},"
-        );
+        let log = LogEvent::from_iter([("message", "hello world")].iter().map(|&s| s));
+        let json = sink.encode_event(Event::from(log)).unwrap();
+        let body = serde_json::to_string(&json).unwrap();
+        assert_eq!(body, "{\"jsonPayload\":{\"message\":\"hello world\"}}");
```

Comment:
So encode event now only encodes one event at a time, but we use the json! macro to ensure that it is correct json. So at compile time we can guarantee that it will generate the correct structure and encode correctly into json. So I think that since we only produce valid Value values via the macro and the macro has been tested it doesn't make much sense to check that here? That said, we could add a test around the build_request phase which I'll add!

Reply:
Probably doesn't make much sense to check that here, yes.

Reply:
I added a new test that should cover more!

https://github.com/timberio/vector/pull/2075/commits/c07c1a1157ebe23ac9b6a6a57a1c41be32c56b7f